DOC convert a few more references to bibtex

romanlutz opened the pull request:
<!--- If relevant, make sure to add a description of your changes in docs/user_guide/installation_and_version_guide/v<major>.<minor>.<patch>.rst -->

## Description
<!--- Provide a general summary of your changes. -->
<!--- Mention related issues, pull requests, or discussions with #<issue/PR/discussion ID>. -->
<!--- Tag people for whom this PR may be of interest using @<username>. -->

Continuing with #1069. I didn't quite know whether the remaining ones at the bottom of the page should stay as is or also get converted to bibtex. I can't find existing bibtex entries for them since they're just blog posts etc.

I also just realized we don't always use `:footcite:` but sometimes additional specifications like `:footcite:p:`. If you have suggestions/thoughts please submit comments or directly commit them. @MiroDudik 

I would also love to order the bibtex file by alias, but that will have to happen in a separate PR for readability reasons.

## Tests
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation
<!--- Select all that apply. -->
- [ ] no documentation changes needed
- [x] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

## Screenshots
<!--- If your PR makes changes to visualizations (e.g., matplotlib code) or the website, please include a screenshot of the result. -->

hildeweerts commented:
>  I can't find existing bibtex entries for them since they're just blog posts etc.

For the sake of consistency, we could consider adding simple `@misc` BibTeX entries for blog posts etc. with the title, URL, and date last accessed.

romanlutz commented:
There are still a whole lot more references to convert, but I'd like to keep this PR at a manageable size. @fairlearn/fairlearn-maintainers can you review and let me know if it's ready to merge?

riedgar-ms commented on docs/refs.bib:
```diff
@@ -184,6 +208,14 @@ @inproceedings{jacobs2021measurement
 }
 
+@misc{kain1975housing,
+  title={Housing markets and racial discrimination: A microeconomic},
```
These title is feeling truncated

romanlutz replied:
bibtex file was bad, fixed it now

riedgar-ms commented on examples/plot_credit_loan_decisions.py:
```diff
@@ -66,17 +65,17 @@
 # other hand, some fairness interventions may be illegal under
-# anti-discrimination law. `Xiang and Raji (2019)
-# <https://arxiv.org/abs/1912.00761>`_ discuss the compatibilities and
+# anti-discrimination law. Xiang and Raji (2019)
```
This is where you want to use the different `:footcite:` modes, to automatically generate the authors and year.

riedgar-ms commented on examples/plot_credit_loan_decisions.py:
```diff
@@ -410,7 +407,7 @@ def resample_training_data(X_train, Y_train, A_train):
 #
 # A secondary harm that is somewhat unique to credit lending decisions is the
 # long-term impact on an individual's credit score. In the United States, a
-# `FICO credit score <https://www.investopedia.com/terms/c/credit_score.asp>`_
```
Why is this being changed to a citation? The link seems more appropriate.

riedgar-ms commented on examples/plot_credit_loan_decisions.py:
```diff
@@ -652,12 +648,13 @@ def plot_group_metrics_with_error_bars(metricframe, metric, error_name):
-# feature* group. The :code:`ThresholdOptimizer` optimizes a specified objective
-# metric (in our case, :code:`balanced_accuracy`) subject to some fairness constraint
+# the :code:`ThresholdOptimizer` algorithm, following Hardt, Price, and Srebro
+# (2016) :footcite:`hardt2016equality`.
```
Again, use one of the other `:footcite:` modes to generate the author list and year automatically.

riedgar-ms commented on examples/plot_credit_loan_decisions.py:
```diff
@@ -1244,7 +1240,7 @@ def create_metricframe_w_errors(mframe, metrics_to_report, metric_error_pair):
 #
 # A key facet of Responsible Machine Learning is responsible documentation
-# practices. `Mitchell et al. (2019) <https://arxiv.org/abs/1810.03993>`_
+# practices. Mitchell et al. (2019) :footcite:`mitchell2019model`
```
Another case for having BibTeX do what it's supposed to do....

riedgar-ms commented on examples/plot_credit_loan_decisions.py:
```diff
@@ -1263,6 +1259,12 @@ def create_metricframe_w_errors(mframe, metrics_to_report, metric_error_pair):
 # with proving their models are compliant with fair lending laws by imputing
-# these demographics features. However, `Chen et. al (2018)
-# <https://arxiv.org/abs/1811.11154>`_ show these imputation methods often
+# these demographics features. However, Chen et. al (2018)
```
And another.

riedgar-ms commented on docs/refs.bib:
```diff
@@ -285,4 +317,127 @@ @inproceedings{barocas2017problem
+
+@misc{scikitlego2019fairness,
+  title={Fairness},
+  author={Vincent Warmerdam, Matthijs Brouns, and Scikit-Lego contributors},
```
I'm not sure this is how BibTeX expects the author list to be formatted. See some of the other entries

riedgar-ms reviewed (changes requested):
I'm not sure all these citation conversions are appropriate, and the other `:footcite:` modes should be being used when "Author (year)" citations are in use.

romanlutz commented:
> I'm not sure all these citation conversions are appropriate, and the other `:footcite:` modes should be being used when "Author (year)" citations are in use.

I've addressed all your comments (I think). Please take another look whenever you have time 🙂

riedgar-ms commented on docs/user_guide/datasets/boston_housing_data.rst:
```diff
@@ -402,47 +412,17 @@ different types of statistical analyses and different predicted outcomes.
 
-  .. [#11] Kinmberlé Crenshaw, Mapping the margins: Intersectionality, identity politics, and violence against women of color,
-      Stanford Law Review, 1993, 43(6), 1241-1299.
+.. [#1] Marshall Hargrave, `"Hedonic Pricing" <https://www.investopedia.com/terms/h/hedonicpricing.asp>`_,
```
Why not just leave the link in the run of the text? I'm not sure why this is being drawn out as a specific citation.

romanlutz replied:
I must have been absent-minded... that's what I wanted to revert actually... I totally agree
